src/java.base/share/classes/java/util/regex/Matcher.java

```diff
@@ -349,7 +356,7 @@ public String group(int group) {
             checkGroup(group);
             if ((groups[group * 2] == -1) || (groups[group * 2 + 1] == -1))
                 return null;
-            return text.subSequence(groups[group * 2], groups[group * 2 + 1]).toString();
+            return text.subSequence(groups[group * 2] - offset, groups[group * 2 + 1] - offset).toString();
```

Comment:
Could be simplified to `return text.substring(groups[group * 2] - offset, groups[group * 2 + 1] - offset);`